MDANSE/Src/MDANSE/Mathematics/Arithmetic.py

```diff
@@ -109,24 +108,34 @@ def adjust_weights(
 
     weights = {}
 
-    cartesianProduct = itertools.product(props, repeat=dim)
-    for elements in cartesianProduct:
-        atom_conc_product = np.prod([contents[el] / n_atms for el in elements])
-        property_product = np.prod(np.array([props[el] for el in elements]), axis=0)
+    if dim == 1:
+        for el in props:
+            atom_conc_product = contents[el] / n_atms
+            property_product = props[el]
```

Comment:
It's saying that you should iterate over `props.items()`
```suggestion
        for el, property_product in props.items():
            atom_conc_product = contents[el] / n_atms
```